src/testers.jl

```diff
@@ -186,44 +185,42 @@ function test_rrule(
         ȳ = output_tangent isa Auto ? rand_tangent(y) : output_tangent
 
         check_inferred && _test_inferred(pullback, ȳ)
-        ∂s = pullback(ȳ)
-        ∂s isa Tuple || error("The pullback must return (∂self, ∂args...), not $∂s.")
-        ∂self = ∂s[1]
-        x̄s_ad = ∂s[2:end]
-        @test ∂self === NoTangent()  # No internal fields
-        msg = "The pullback should return 1 cotangent for each primal input."
-        @test_msg msg length(x̄s_ad) == length(args)
+        ad_cotangents = pullback(ȳ)
+        ad_cotangents isa Tuple || error("The pullback must return (∂self, ∂args...), not $∂s.")
+        msg = "The pullback should return 1 cotangent for the primal and each primal input."
+        @test_msg msg length(ad_cotangents) == 1 + length(args)
 
         # Correctness testing via finite differencing.
         # TODO: remove Nothing when https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/113
-        x̄s_is_dne = isa.(accumulated_x̄, Union{Nothing,NoTangent})
-        if any(accumulated_x̄ .== nothing)
+        is_ignored = isa.(accum_cotangents, Union{Nothing, NoTangent})
+        if any(accum_cotangents .== nothing)
             Base.depwarn(
                 "test_rrule(f, k ⊢ nothing) is deprecated, use " *
                 "test_rrule(f, k ⊢ NoTangent()) instead for non-differentiable ks",
                 :test_rrule,
             )
         end
 
-        x̄s_fd = _make_j′vp_call(fdm, (xs...) -> f(xs...; fkwargs...), ȳ, xs, x̄s_is_dne)
-        for (accumulated_x̄, x̄_ad, x̄_fd) in zip(accumulated_x̄, x̄s_ad, x̄s_fd)
-            if accumulated_x̄ isa Union{Nothing,NoTangent}  # then we marked this argument as not differentiable # TODO remove once #113
-                @assert x̄_fd === nothing  # this is how `_make_j′vp_call` works
-                x̄_ad isa ZeroTangent && error(
+        fd_cotangents = _make_j′vp_call(fdm, (f, xs...) -> f(xs...; fkwargs...), ȳ, (f, xs...), is_ignored)
```

Comment:
This is the main change, i.e. from
`(xs...) -> f(xs...; fkwargs...)` to
`(f, xs...) -> f(xs...; fkwargs...)`
making the `f` an argument. The rest of the changes are adding an extra cotangent for `f` via `rand_tangent`, and renaming the variables from `x̄_something` to `something_cotangent` since it also includes `f̄`